controller/v1/user.go

```diff
@@ -421,10 +421,16 @@ func (uc *UserController) RedirecToSericeAuth(c echo.Context) error {
 
 	logger.Debug("Doing URL for provider:", providerName)
 
-	u := config.Setting.MAIN_SETTINGS.OAuth2Config.AuthCodeURL(config.Setting.OAUTH2_SETTINGS.StateValue,
-		oauth2.SetAuthURLParam("response_type", config.Setting.OAUTH2_SETTINGS.ResponseType),
-		oauth2.SetAuthURLParam("code_challenge", heputils.GenCodeChallengeS256(config.Setting.OAUTH2_SETTINGS.UserToken)),
-		oauth2.SetAuthURLParam("code_challenge_method", "S256"))
+	u := ""
```

Comment:
```

options := []oauth2.AuthCodeOption{
	oauth2.SetAuthURLParam("response_type", config.Setting.OAUTH2_SETTINGS.ResponseType),
}

if  config.Setting.OAUTH2_SETTINGS.UsePkce {
	options = append(options, oauth2.SetAuthURLParam("redirect_uri", redirectUrl),
		oauth2.SetAuthURLParam("code_challenge", heputils.GenCodeChallengeS256(config.Setting.OAUTH2_SETTINGS.UserToken)),
		oauth2.SetAuthURLParam("code_challenge_method", "S256"))
}

u :=  config.Setting.MAIN_SETTINGS.OAuth2Config.AuthCodeURL(config.Setting.OAUTH2_SETTINGS.StateValue, options...)
```

Reply:
Probably this way will be more elegance ?

Reply:
Modified as requested

Reply:
Modified as requested